RELEASE_NOTES.rst

```diff
@@ -19,8 +19,123 @@
+- Fix execution API server URL handling for relative paths in KE (#51183)
+- Add log lookup exception for Empty operator subtypes (#50325)
+- Increase the max zoom on the graph view to make it easier to see small dags on big monitor screens (#50772)
+- Fix timezone selection and dashboard layout (#50463)
+- Bugfix: Creating backfill for a dag is affecting other dags (#50577)
+- Fix next asset schedule and dag card UX (#50271)
+- Add bundle path to sys.path in task runner (#51318)
+- Add bundle root to sys.path in dag processor (#50385)
+- Prevent CPU spike in task supervisor when heartbeat timeout exceeded (#51023)
+- Fix Airflow Connection Form widget error (#51168)
+- Add backwards compatibility shim and deprecation warning for EmailOperator (#51004)
+- Handle ``SIGSEGV`` signals during DAG file imports (#51171)
+- Fix deferred task resumption in ``dag.test()`` (#51182)
+- Fix get dags query to not have join explosion (#50984)
+- Bugfix: Logical date isn't populated in Context vars (#50898)
+- Mask variable values in task logs only if the variable key is sensitive (#50775)
+- Mask secrets when retrieving variables from secrets backend (#50895)
+- Deserialize should work while retrieving variables with secrets backend (#50889)
+- Fix XCom deserialization for mapped tasks with custom backend (#50687)
+- Support macros defined via plugins in Airflow 3 (#50642)
+- Fix Pydantic ``ForwardRef`` error by reordering discriminated union definitions (#50688)
+- Adding backwards compatibility shim for ``BaseNotifier`` (#50340)
+- Use latest bundle version when clearing / re-running dag (#50040)
+- Handle ``upstream_mapped_index`` when xcom access is needed (#50641)
+- Remove unnecessary breaking flag in config command (#50781)
+- Do not flood worker logs with secrets backend loading logs (#50581)
+- Persist table sorting preferences across sessions using local storage (#50720)
+- Fixed patch_task_instance API endpoint to support task instance summaries and task groups (#50550)
+- Fixed bulk API schemas to improve OpenAPI compatibility and client generation (#50852)
+- Fixed variable API endpoints to support keys containing slashes (#50841)
+- Restored backward compatibility for the ``/run`` API endpoint for older Task SDK clients
+- Fixed dropdown overflow and error text styling in ``FlexibleForm`` component (#50845)
+- Corrected DAG tag rendering to display ``+1 more`` when tags exceed the display limit by one (#50669)
+- Fix permission check on the ui config endpoint (#50608)
+- Fix ``default_args`` handling in operator ``.partial()`` to prevent ``TypeError`` when unused keys are present (#50525)
+- DAG Processor: Fix index to sort by last parsing duration (#50388)
+- UI: Fix border overlap issue in the Events page (#50453)
+- Fix ``airflow tasks clear`` command (#49631)
+- Restored support for ``--local`` flag in ``dag list`` and ``dag list-import-errors`` CLI commands (#49380)
```

Comment:
This should probably be a 3.1 thing too. We didn't have a local flag before - that was just the way it worked.

``airflow dags reserialize`` is enough, you don't need to run a dag processor.

Reply:
The difference though is folks utilized it earlier in 2.x to verify if the dag files have any error locally. As an example check https://airflow.apache.org/docs/apache-airflow/stable/faq.html#why-is-task-not-getting-scheduled

>Does your script “compile”, can the Airflow engine parse it and find your DAG object? To test this, you can run airflow dags list and confirm that your DAG shows up in the list. You can also run airflow dags show foo_dag_id and confirm that your task shows up in the graphviz format as expected. If you use the CeleryExecutor, you may want to confirm that this works both where the scheduler runs as well as where the worker runs.

Happy to remove it if you feel strongly about it.

Reply:
Kaxil and I had a chat, we will leave this in.